types/src/zkid.rs

```diff
@@ -225,13 +229,45 @@ impl Claims {
     }
 }
 
-pub type G1 = Vec<String>;
-pub type G2 = Vec<Vec<String>>;
-#[derive(Clone, Debug, Deserialize, PartialEq, Eq, Hash, Serialize)]
+#[derive(
+    Clone, Debug, Deserialize, PartialEq, Eq, Hash, Serialize, CryptoHasher, BCSCryptoHash,
+)]
 pub struct Groth16Zkp {
-    a: G1,
-    b: G2,
-    c: G1,
+    a: G1Projective,
+    b: G2Projective,
+    c: G1Projective,
+}
+
+#[derive(Clone, Debug, Deserialize, PartialEq, Eq, Hash, Serialize)]
+pub struct SignedGroth16Zkp {
+    pub proof: Groth16Zkp,
+    /// The signature of the proof signed by the private key of the `ephemeral_pubkey`.
+    pub proof_signature: EphemeralSignature,
```

Comment:
Is this the non-malleability signature or the training wheels signature?

Reply:
non-malleability signature

Reply:
Okay. I renamed.